### PR TITLE
Add show method for CATKE closure

### PR DESCRIPTION
**Pull request by navidcy**
### before this PR

```Julia
julia> using Oceananigans

julia> using Oceananigans.TurbulenceClosures.CATKEVerticalDiffusivities: CATKEVerticalDiffusivity

julia> catke = CATKEVerticalDiffusivity()
┌ Warning: CATKEVerticalDiffusivity is an experimental turbulence closure that
│ is unvalidated and whose default parameters are not calibrated for
│ realistic ocean conditions or for use in a three-dimensional
│ simulation. Use with caution and report bugs and problems with physics
│ to https://github.com/CliMA/Oceananigans.jl/issues.
└ @ Oceananigans.TurbulenceClosures.CATKEVerticalDiffusivities /g/data/v45/nc3020/.julia/packages/Oceananigans/Bks9B/src/TurbulenceClosures/turbulence_closure_implementations/CATKEVerticalDiffusivities/CATKEVerticalDiffusivities.jl:95
CATKEVerticalDiffusivity{Oceananigans.TurbulenceClosures.VerticallyImplicitTimeDiscretization, Float64, Oceananigans.TurbulenceClosures.CATKEVerticalDiffusivities.MixingLength{Float64}, Oceananigans.TurbulenceClosures.CATKEVerticalDiffusivities.SurfaceTKEFlux{Float64}}(2.91, Oceananigans.TurbulenceClosures.CATKEVerticalDiffusivities.MixingLength{Float64}(1.16, 0.5, 0.5, 0.5, 0.0, 0.0, 0.0, 0.15, 3.87, 0.4, 0.77, 0.13, 1.11, 0.72, 0.76), Oceananigans.TurbulenceClosures.CATKEVerticalDiffusivities.SurfaceTKEFlux{Float64}(3.62, 1.31))
```

### after this PR

```julia
julia> using Oceananigans
[ Info: Oceananigans will use 48 threads

julia> using Oceananigans.TurbulenceClosures.CATKEVerticalDiffusivities: CATKEVerticalDiffusivity

julia> catke = CATKEVerticalDiffusivity()
┌ Warning: CATKEVerticalDiffusivity is an experimental turbulence closure that
│ is unvalidated and whose default parameters are not calibrated for
│ realistic ocean conditions or for use in a three-dimensional
│ simulation. Use with caution and report bugs and problems with physics
│ to https://github.com/CliMA/Oceananigans.jl/issues.
└ @ Oceananigans.TurbulenceClosures.CATKEVerticalDiffusivities /g/data/v45/nc3020/.julia/packages/Oceananigans/Bks9B/src/TurbulenceClosures/turbulence_closure_implementations/CATKEVerticalDiffusivities/CATKEVerticalDiffusivities.jl:95
CATKEVerticalDiffusivity:
(Cᴰ=2.91,
 MixingLength: (Cᴸᵇ=1.16,
                Cᵟu=0.5, Cᵟc=0.5, Cᵟe=0.5,
                Cᴬu=0.0, Cᴬc=0.0, Cᴬe=0.0,
                Cᴷu⁻=0.15, Cᴷc⁻=0.4, Cᴷe⁻=0.13,
                Cᴷuʳ=3.87, Cᴷcʳ=0.77, Cᴷeʳ=1.11,
                CᴷRiʷ=0.72, CᴷRiᶜ=0.76),
 SurfaceTKEFlux: (Cᵂu★=3.62, CᵂwΔ=1.31))
```

### CODE REVIEW

**Review by glwagner — APPROVED** [review]
Nice!

Tabular view might be slightly better but this is an excellent start.

**Comment by navidcy** [comment]
Yes. We also need to show the timestepping method...

**Comment by navidcy** [comment]
What's "tabular view"?

**Comment by glwagner** [comment]
I guess making a table (rather than showing three parameters on one line), something like

```
CATKEVerticalDiffusivity{VerticallyImplicitTimeDiscretization} with parameters
    Cᴰ = 2.91
    MixingLength:
        Cᴸᵇ = 1.16,
        Cᵟu = 0.5
        Cᵟc = 0.5
<etc>

**Comment by navidcy** [comment]
After  13520fd

```Julia
julia> catke
CATKEVerticalDiffusivity with Oceananigans.TurbulenceClosures.VerticallyImplicitTimeDiscretization and parameters:
    Cᴰ=2.91,
    MixingLength:
        Cᴸᵇ=1.16,
        Cᵟu=0.5,
        Cᵟc=0.5,
        Cᵟe=0.5,
        Cᴬu=0.0,
        Cᴬc=0.0,
        Cᴬe=0.0,
        Cᴷu⁻=0.15,
        Cᴷc⁻=0.4,
        Cᴷe⁻=0.13,
        Cᴷuʳ=3.87,
        Cᴷcʳ=0.77,
        Cᴷeʳ=1.11,
        CᴷRiʷ=0.72,
        CᴷRiᶜ=0.76,
    SurfaceTKEFlux:
        Cᵂu★=3.62,
        CᵂwΔ=1.31
```

File: src/TurbulenceClosures/turbulence_closure_implementations/CATKEVerticalDiffusivities/CATKEVerticalDiffusivities.jl [hunk 1]
```diff
@@ -287,11 +287,11 @@ end
...
 ##### Show
 #####
-Base.show(io::IO, closure::CATKEVD) =
-    print(io, "CATKEVerticalDiffusivity: \n" *
-              "(Cᴰ=$(closure.Cᴰ), \n" * 
-              " $(closure.mixing_length), \n" *
-              " $(closure.surface_TKE_flux)" * ")")
+Base.show(io::IO, closure::CATKEVD{TD}) where TD =
+    print(io, "CATKEVerticalDiffusivity with $(TD) and parameters: \n" *
+              "    Cᴰ=$(closure.Cᴰ), \n" *
```
Comment (glwagner) [inline src/TurbulenceClosures/turbulence_closure_implementations/CATKEVerticalDiffusivities/CATKEVerticalDiffusivities.jl:292]:
```suggestion
              "    Cᴰ = $(closure.Cᴰ), \n" * 
```

**Comment by navidcy** [comment]
OK, now:

```julia
julia> using Oceananigans
[ Info: Precompiling Oceananigans [9e8cae18-63c1-5223-a75c-80ca9d6e9a09]
[ Info: Oceananigans will use 12 threads

julia> using Oceananigans.TurbulenceClosures.CATKEVerticalDiffusivities: CATKEVerticalDiffusivity

julia> catke = CATKEVerticalDiffusivity()
┌ Warning: CATKEVerticalDiffusivity is an experimental turbulence closure that
│ is unvalidated and whose default parameters are not calibrated for
│ realistic ocean conditions or for use in a three-dimensional
│ simulation. Use with caution and report bugs and problems with physics
│ to https://github.com/CliMA/Oceananigans.jl/issues.
└ @ Oceananigans.TurbulenceClosures.CATKEVerticalDiffusivities ~/Research/OC.jl/src/TurbulenceClosures/turbulence_closure_implementations/CATKEVerticalDiffusivities/CATKEVerticalDiffusivities.jl:96
CATKEVerticalDiffusivity with Oceananigans.TurbulenceClosures.VerticallyImplicitTimeDiscretization and parameters:
    Cᴰ = 2.91,
    MixingLength:
          Cᴸᵇ = 1.16,
          Cᵟu = 0.5,
          Cᵟc = 0.5,
          Cᵟe = 0.5,
          Cᴬu = 0.0,
          Cᴬc = 0.0,
          Cᴬe = 0.0,
         Cᴷu⁻ = 0.15,
         Cᴷc⁻ = 0.4,
         Cᴷe⁻ = 0.13,
         Cᴷuʳ = 3.87,
         Cᴷcʳ = 0.77,
         Cᴷeʳ = 1.11,
        CᴷRiʷ = 0.72,
        CᴷRiᶜ = 0.76,
    SurfaceTKEFlux:
         Cᵂu★ = 3.62,
         CᵂwΔ = 1.31
```

**Pull request merged**